build(chart): support for externally managed secret

davekonopka opened the pull request:
## What? (description)

Add option to Helm chart to use an externally managed secret.

## Why? (reasoning)

I want to be able to use external secrets to manage my secret from an external source like AWS Secrets Manager rather than have the values passed through chart values. It defaults to the current functionality of creating a secret.

## Screenshots (if applicable)

Confirmed the functionality by installing a release with values:

```
  secret:
    createSecret: false
    secretName: slackmoji-notifier-private
```

And seeing deployment get connected to an external secret:

```
      envFrom:
        - secretRef:
            name: slackmoji-notifier-private
```

## GitHub Issue (if applicable)


## Acceptance
Check your PR for the following:

- [ ] you included tests
- [x] you linted your code
- [x] your PR has appropriate, atomic commits (interactive rebase!)
- [x] your commit message follows Conventional Commit format
- [x] you are not reducing the total test coverage

davekonopka-gametime committed:
build(chart): support for externally managed secret

particledecay commented on chart/templates/deployment.yaml:
```diff
@@ -45,7 +45,7 @@ spec:
           envFrom:
             - secretRef:
-                name: {{ include "slackmoji-notifier.fullname" . }}
+                name: {{ .Values.secret.secretName | default (include "slackmoji-notifier.fullname" .) }}
```
Love this change, you caught me slipping

davekonopka-gametime committed:
build(chart): bump version to 0.4.0

davekonopka commented:
@particledecay Chart is bumped to `0.3.0` I went with minor bump even though values structure changed since it's still pre v1.

particledecay commented:
sorry I went with the other one first but can you update this one now? :grimacing:

davekonopka commented:
@particledecay I bumped the chart version but realized the app version has updated from the other PR. I'm guessing the app will update again after this merges since it's a feat? Should I make the title a chore and include bumping the app version to version 0.4.0 in the chart?

davekonopka-gametime committed:
build(chart): update app version to latest

davekonopka commented:
@particledecay I updated the commits to `build:` and updated the app version and image tag value to latest release. The chart and app version happen to line up but not intentionally.

davekonopka commented:
@particledecay Just checking in to see if I can do anything else for this PR.
